docs/sources/operators-guide/reference-http-api/index.md

```diff
@@ -683,13 +683,28 @@ POST /api/v1/rules/{namespace}
 POST <prometheus-http-prefix>/rules/{namespace}
 ```
 
-Creates or updates a rule group. This endpoint expects a request with `Content-Type: application/yaml` header and the
-rules **YAML** definition in the request body, and returns `202` on success.
+Creates or updates a rule group.
+This endpoint expects a request with `Content-Type: application/yaml` header and the rules group **YAML** definition in the request body, and returns `202` on success.
+The request body must contain the definition of one and only one rule group.
 
 This endpoint can be disabled via the `-ruler.enable-api` CLI flag (or its respective YAML config option).
 
 Requires [authentication](#authentication).
 
+> **Note:** When using `curl` send the request body from a file, ensure that you use the `--data-binary` flag instead of `-d`, `--data`, or `--data-ascii`.
+> The latter options do not preserve carriage returns and newlines.
+
+#### Example request body
```

Comment:
Awesome. I think the example makes the expected structure of HTTP payload very clear.